src/sdam/topology_description.ts

```diff
@@ -431,7 +431,12 @@ function updateRsWithPrimaryFromMember(
   setName?: string
 ): TopologyType {
   if (setName == null) {
-    throw new MongoDriverError('setName is required');
+    // FIXME(NODE-3402): Consider replacing with MongoCompatibilityError or making
+    // error message more verbose to indicate that the incompatibility is occurring
+    // with the topology rather than the current server version
+    throw new MongoCompatibilityError(
```

Comment:
I think this is a runtime error, this means we failed to acquire or pass in the setName, it's not user configuration specific